### ISSUE TITLE
Child class setUpTestData is not called when time_machine is used as a class decorator for Django tests

### ISSUE THREAD

**Issue by vkhudiakova** [adamchainz/time-machine#402]
### Python Version

3.11

### Package Version

2.13.0

### Description

Here is a repository with example: https://github.com/vkhudiakova/time-machine-bug
If I use `time_machine` as a class decorator for the base class, that is inherited from Django TestCase, then the child `setUpTestData` method is not called.

```
from datetime import datetime

import time_machine
from django.test import TestCase


@time_machine.travel(datetime(2023, 1, 1))
class BaseTest(TestCase):
    @classmethod
    def setUpTestData(cls):
        super().setUpTestData()
        print('Base class method')


class ChildTest(BaseTest):
    @classmethod
    def setUpTestData(cls):
        super().setUpTestData()
        print('Child class method')

    def test_something(self):
        self.assertTrue(False)
```

**Expected result:**
Test outputs:

```
Base class method
Child class method
```
Running both base and child classes' setUpTestData methods.

**Actual result**

Test outputs only `Base class method`

If @time_machine decorator is removed from the class, the test works as expected.

**Comment by adamchainz** [comment]
Aww dammit. Class decorators are hard to get right.

Can you try recreating this in a test case in time-machine’s test suite? Rather than using `setUpTestData`, since there’s no Django in the project, you should be able to get the same effect with `setUpClass`. (Django’s `setUpClass` calls `setUpTestData`.) Run the tests with `tox`, such as `tox -e py311` for Python 3.11.

**Comment by vkhudiakova** [comment]
@adamchainz Thank you for the quick response and for the library :)
I'll take a look at the test suite.

**Comment by bp72** [comment]
Hi! I added a small failing test, I hope it'll help!
https://github.com/adamchainz/time-machine/pull/404

**Comment by adamchainz** [comment]
Fixed in #404 which has been released.

**Comment by vkhudiakova** [comment]
Thank you for the fix!
